examples/machine_learning/src/lib.rs

```diff
@@ -138,63 +138,111 @@ fn evaluate_prediction(hits: &mut u32, animal: &Animal, prediction: &[f64]) -> (
     (actual_type, accurate)
 }
 
-struct Node;
+struct Node {
+    training_set_size: u32,
+    test_set_size: u32,
+    training_matrix: Option<Matrix<f64>>,
+    target_matrix: Option<Matrix<f64>>,
+    test_matrix: Option<Matrix<f64>>,
+    test_animals: Option<Vec<Animal>>,
+    model: NaiveBayes<naive_bayes::Gaussian>,
+}
 
 impl oak::Node for Node {
     fn new() -> Self {
-        Node
+        Node {
+	    training_set_size: 1000,
+	    test_set_size: 1000,
+	    training_matrix: None,
+	    target_matrix: None,
+	    test_matrix: None,
+	    test_animals: None,
+	    model: NaiveBayes::<naive_bayes::Gaussian>::new(),
```

Comment:
The compiler should be able to infer the correct generic type here, does it not? i.e. `NaiveBayes::new()` should work, since the target field already has an unambiguous monomorphic type.

Reply:
Done.